src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -203,15 +208,29 @@ private async Task UpdateJobRecordAsync(CancellationToken cancellationToken)
                 SearchResult searchResult = null;
 
                 // Search and process the results.
-                using (IScoped<ISearchService> searchService = _searchServiceFactory())
+                switch (_exportJobRecord.ExportType)
                 {
-                    searchResult = await searchService.Value.SearchAsync(
-                        resourceType: null,
-                        queryParametersList,
-                        cancellationToken);
+                    case ExportJobType.All:
+                    case ExportJobType.Patient:
+                        using (IScoped<ISearchService> searchService = _searchServiceFactory())
+                        {
+                            searchResult = await searchService.Value.SearchAsync(
+                                resourceType: null,
+                                queryParametersList,
+                                cancellationToken);
+                        }
+
+                        break;
+                    case ExportJobType.Group:
+                        searchResult = await GetGroupPatients(
+                            _exportJobRecord.GroupId,
+                            queryParametersList,
+                            _exportJobRecord.QueuedTime,
+                            cancellationToken);
+                        break;
                 }
 
-                if (_exportJobRecord.ExportType == ExportJobType.Patient)
+                if (_exportJobRecord.ExportType == ExportJobType.Patient || _exportJobRecord.ExportType == ExportJobType.Group)
```

Comment:
In the below code, when we call `ProcessProgressChange` do we need to update the `forceCommit` flag to be set to true for `Group` export also?